File: Chapter 8/TestShop.cs
using System;

namespace TestShop
{
    class TestShop
    {

        public static void Main()
        {

            Shop shop = null;

            Console.WriteLine("Instantiate the PizzaShop class:" + "\n");

            shop = new PizzaShop();

            string[] inventory = shop.getInventory();


            // list the inventory

            for (int i = 0; i < 5; i++)
            {
                Console.WriteLine("Argument" + i + " = " + inventory[i]);
            }


            // buy an item

            shop.buyInventory(inventory[1]);

        }

    }



    public abstract class Shop  {

        public void CalculateSaleTax() {

            Console.WriteLine("Calculate Sales Tax");

        }

        public abstract string[] getInventory();

        public abstract void buyInventory(string item);

    }

    public interface Nameable {

          string getName();
          void setName(string name);

    }

    public class PizzaShop : Shop , Nameable 
    {

        string _CompanyName;

        string[] foodOfferings = {
            "Pizza",
            "Spaghetti",
            "Garden Salad",
            "Anitpasto",
            "Calzone"
        };

        public override string[] getInventory() {

            return foodOfferings;

        }

        public override void buyInventory(string item) {

            Console.WriteLine("\nYou have just purchased " + item);

        }

        public string getName(){

            return _CompanyName;
        }

        public void setName(string name){

            _CompanyName = name;
        }

    }

    public class DonutShop : Shop , Nameable {

        string _CompanyName;

        string[] menuItems = {
            "Donuts",
            "Muffins",
            "Danish",
            "Coffee",
            "Tea"
        };

        public override string[]  getInventory() {

            return menuItems;

        }

        public override void  buyInventory(string item) {

            Console.WriteLine(string.format("\nYou have just purchased {0}.", item);

        }

        public string getName(){

            return _CompanyName;
        }

        public void setName(string name){

            _CompanyName = name;
        }
    }

}

File: Chapter 8/TestShop.java
class TestShop {

   public static void main (String args[]) {

      Shop shop = null;

      String className = "DonutShop";

      System.out.println("Instantiate the class:" + className + "\n");


      try {

	   // new pizzaShop();
       shop = (Shop)Class.forName(className).newInstance();

      } catch (Exception e) {

         e.printStackTrace();
      }

      String[] inventory = shop.getInventory();


      // list the inventory

      for (int i=0; i<inventory.length; i++) {
         System.out.println("Argument" + i + " = " + inventory[i]);
      }


      // buy an item

      shop.buyInventory(inventory[1]);

   }

}

public abstract class Shop  {

    CustList customerList;

    public void CalculateSaleTax() {

        System.out.println("Calculate Sales Tax");

    };

    public abstract String[] getInventory();

    public abstract void buyInventory(String item);

}

public interface Nameable {

    public abstract String getName();
    public abstract void setName(String name);

}

public class PizzaShop extends Shop implements Nameable {

    String companyName;


    String[] foodOfferings = {
        "Pizza",
        "Spaghetti",
        "Garden Salad",
        "Anitpasto",
        "Calzone"
    };

    public String[] getInventory() {

        return foodOfferings;

    }

    public void buyInventory(String item) {

        System.out.println("\nYou have just purchased " + item);

    }

    public String getName(){

        return companyName;
    }

    public void setName(String name){

        companyName = name;
    }

}

public class DonutShop extends Shop implements Nameable {

    String companyName;

    String[] menuItems = {
        "Donuts",
        "Muffins",
        "Danish",
        "Coffee",
        "Tea"
	};

    public String[] getInventory() {

        return menuItems;

    }

    public void buyInventory(String item) {

        System.out.println("\nYou have just purchased " + item);

    }

    public String getName(){

        return companyName;
    }

    public void setName(String name){

        companyName = name;
    }
}

public class CustList {

    String name;

    public  String findCust() {return name;}
    public  void addCust(String Name){}

}
